src/plugins/rhbz.c

```diff
@@ -874,3 +890,161 @@ xmlrpc_value *rhbz_search_duphash(struct abrt_xmlrpc *ax,
 
     return bugs;
 }
+
+xmlrpc_value *rhbz_get_sub_components(struct abrt_xmlrpc *ax, const char *product, const char *component)
+{
+    xmlrpc_env env;
+    xmlrpc_env_init(&env);
+
+    /* Product info is stored in a struct as follows:
+    {"products":({"id":i,"name":s,"description":s,"is_active":b,"default_milestone":s,"default_release":s,"has_unconfirmed":b,"classification":s,"components":({"id":i,"name":s,"description":s,"default_assigned_to":s,"default_qa_contact":s,"default_docs_contact":s,"default_cc":(s),"sort_key":i,"is_active":b,"sub_components":({})})})}
+
+    where:
+    () = array
+    {} = struct
+    b  = bool
+    i  = int
+    s  = string
+
+    The structs in the "sub_components" array are almost identical to the ones in the parent
+    "components" array. The same is repeated recursively for subcomponents that themselves contain
+    subcomponents.
+
+    http://xmlrpc-c.sourceforge.net/doc/libxmlrpc.html#formatstring
+
+    First, obtain a struct containing one item called "products" whose value is an array containing one struct
+    containing information about the product we specified in the function arg. The "include_fields" arg
+    specifies the struct as the following subset of the above:
+    {"products":({"name":s,"components":({"name":s,"sub_components":({})})})}
+
+    The "components" struct doesn't have to be specified by itself because we specify its members.
+    For some reason, the same doesn't work for "components.sub_components". */
+
+    xmlrpc_value *products_array = abrt_xmlrpc_call(ax, "Product.get", "{s:(s),s:(ssss)}",
+                                                    "names", product,
+                                                    "include_fields",
+                                                    "name",
+                                                    "components.name",
+                                                    "components.sub_components",
+                                                    "components.sub_components.name");
+    if (env.fault_occurred)
```

Comment:
How does this interact with Bugzilla instances without the extension?

Reply:
I added a check to make sure we only retrieve the subcomponents for product "Red Hat Enterprise Linux <X>". With product "Fedora", the query for subcomponent fails after a _long_ time of waiting. I wonder if it's possible to find out the info that product so-and-so requires subcomponents directly, without the `Product.get` call.

Reply:
Probably because there is an exorbitant amount of components for the Fedora product. Instead of using hacks, I would try to reach out to the maintainer of Red Hat Bugzilla. Likely this will require using an extension to query a component directly. This will need changing regardless because of that detail.

With the above, the answer to my original question would be “not very”, so we will need handling to not explode should someone use a more conventional Bugzilla instance.

Reply:
So I [wrote to RH BZ maintainers](https://redhat.service-now.com/help?id=rh_ticket&table=sc_req_item&sys_id=a8c8c218db705c90c0c8464e1396192e), (probably using a completely wrong communication channel). Thoughts?